src/include/sof/dma.h

```diff
@@ -160,6 +160,8 @@ struct dma_ops {
 
 	int (*probe)(struct dma *dma);
 	int (*remove)(struct dma *dma);
+
+	int (*get_data_size)(struct dma *dma, int channel);
```

Comment:
The naming here is a little misleading, what we really need to know is the buffer free/avail bytes. Btw, could this not be worked out using the buffer APIs or DMA position API ?

Reply:
In the PR #990 you've proposed get_data_size(), because "avail/free have usage meaning in ALSA". I really don't care about the naming convention, just would like to avoid separating this into two methods (e.g. avail and free), since only one of them will be used per DMA direction.

DMA position is really closer to the total number of transferred bytes and not current r_ptr/w_ptr in the circular buffer, so it would be even more misleading. Buffer API is really mirroring logical position of the pointers and not the real HW position.

Reply:
Ok, so this is the DMA buffer free/avail based on runtime HW position of DMA. It would still be useful to provide free and vail in the API call. i.e.
```
int get_data_size(dma, int channel, uint32_t *free, uint32_t *avail)
```

Reply:
@lgirdwood Changed according to your proposition.